[staging-next] PHP 8.2 and 8.3 fixes for libxml2 2.13

reckenrode opened the pull request:
## Description of changes

Applies upstream patches to fix libxml2 2.13 compatibility issues. Fixes build failures on staging-next https://github.com/NixOS/nixpkgs/pull/328673.

- dom (https://hydra.nixos.org/build/267080459, https://hydra.nixos.org/build/267233673)
- simplexml (https://hydra.nixos.org/build/267172447, https://hydra.nixos.org/build/267258056)
- soap (fixed upstream though passing on Hydra)
- xml (fixed upstream though passing on Hydra)
- xmlwriter (fixed upstream though passing on Hydra)

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

reckenrode committed:
php{82,83}Extensions.dom: apply upstream patch for libxml2 2.13 compatibility

reckenrode committed:
php{83,83}Extensions.simplexml: apply upstream patch for libxml2 2.13 compatibility

reckenrode committed:
php{83,83}Extensions.soap: apply upstream patch for libxml2 2.13 compatibility

reckenrode committed:
php{83,83}Extensions.xml: apply upstream patch for libxml2 2.13 compatibility

reckenrode committed:
php{83,83}Extensions.xmlwriter: apply upstream patch for libxml2 2.13 compatibility

reckenrode commented:
I can drop soap, xml, and xmlwriter if that would be preferred since they do seem to build on Hydra.

reckenrode commented:
Result of `nixpkgs-review pr 330299` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>qcachegrind</li>
  </ul>
</details>
<details>
  <summary>22 packages failed to build:</summary>
  <ul>
    <li>n98-magerun</li>
    <li>php81</li>
    <li>php81Extensions.soap</li>
    <li>php81Extensions.soap.dev</li>
    <li>php81Packages.castor</li>
    <li>php81Packages.composer</li>
    <li>php81Packages.composer-local-repo-plugin</li>
    <li>php81Packages.cyclonedx-php-composer</li>
    <li>php81Packages.deployer</li>
    <li>php81Packages.grumphp</li>
    <li>php81Packages.phan</li>
    <li>php81Packages.phing</li>
    <li>php81Packages.phive</li>
    <li>php81Packages.php-codesniffer</li>
    <li>php81Packages.php-cs-fixer</li>
    <li>php81Packages.php-parallel-lint</li>
    <li>php81Packages.phpinsights</li>
    <li>php81Packages.phpmd</li>
    <li>php81Packages.phpstan</li>
    <li>php81Packages.psalm</li>
    <li>php81Packages.psysh</li>
    <li>snipe-it</li>
  </ul>
</details>
<details>
  <summary>108 packages built:</summary>
  <ul>
    <li>adminer</li>
    <li>adminerevo</li>
    <li>php (apacheHttpdPackages.php)</li>
    <li>composer-require-checker</li>
    <li>davis</li>
    <li>engelsystem</li>
    <li>firefly-iii</li>
    <li>flarum</li>
    <li>frankenphp</li>
    <li>freshrss</li>
    <li>grocy</li>
    <li>icingaweb2</li>
    <li>laravel</li>
    <li>matomo</li>
    <li>matomo-beta</li>
    <li>matomo_5</li>
    <li>movim</li>
    <li>n98-magerun2</li>
    <li>nagios</li>
    <li>nextcloud-news-updater</li>
    <li>nextcloud-news-updater.dist</li>
    <li>nominatim</li>
    <li>paratest</li>
    <li>pdepend</li>
    <li>pest</li>
    <li>phel</li>
    <li>phoronix-test-suite</li>
    <li>php81Packages.box (php82Packages.box ,php83Packages.box)</li>
    <li>php82Extensions.dom</li>
    <li>php82Extensions.dom.dev</li>
    <li>php82Extensions.simplexml</li>
    <li>php82Extensions.simplexml.dev</li>
    <li>php82Extensions.soap</li>
    <li>php82Extensions.soap.dev</li>
    <li>php82Extensions.xml</li>
    <li>php82Extensions.xml.dev</li>
    <li>php82Extensions.xmlreader</li>
    <li>php82Extensions.xmlreader.dev</li>
    <li>php82Extensions.xmlwriter</li>
    <li>php82Extensions.xmlwriter.dev</li>
    <li>php82Extensions.xsl</li>
    <li>php82Extensions.xsl.dev</li>
    <li>php82Extensions.zend_test</li>
    <li>php82Extensions.zend_test.dev</li>
    <li>php82Packages.castor</li>
    <li>php82Packages.composer</li>
    <li>php82Packages.composer-local-repo-plugin</li>
    <li>php82Packages.cyclonedx-php-composer</li>
    <li>php82Packages.deployer</li>
    <li>php82Packages.grumphp</li>
    <li>php82Packages.phan</li>
    <li>php82Packages.phing</li>
    <li>php82Packages.phive</li>
    <li>php82Packages.php-codesniffer</li>
    <li>php82Packages.php-cs-fixer</li>
    <li>php82Packages.php-parallel-lint</li>
    <li>php82Packages.phpinsights</li>
    <li>php82Packages.phpmd</li>
    <li>php82Packages.phpstan</li>
    <li>php82Packages.psalm</li>
    <li>php82Packages.psysh</li>
    <li>php83</li>
    <li>php83Extensions.dom</li>
    <li>php83Extensions.dom.dev</li>
    <li>php83Extensions.simplexml</li>
    <li>php83Extensions.simplexml.dev</li>
    <li>php83Extensions.soap</li>
    <li>php83Extensions.soap.dev</li>
    <li>php83Extensions.xml</li>
    <li>php83Extensions.xml.dev</li>
    <li>php83Extensions.xmlreader</li>
    <li>php83Extensions.xmlreader.dev</li>
    <li>php83Extensions.xmlwriter</li>
    <li>php83Extensions.xmlwriter.dev</li>
    <li>php83Extensions.xsl</li>
    <li>php83Extensions.xsl.dev</li>
    <li>php83Extensions.zend_test</li>
    <li>php83Extensions.zend_test.dev</li>
    <li>php83Packages.castor</li>
    <li>php83Packages.composer</li>
    <li>php83Packages.composer-local-repo-plugin</li>
    <li>php83Packages.cyclonedx-php-composer</li>
    <li>php83Packages.deployer</li>
    <li>php83Packages.grumphp</li>
    <li>php83Packages.phan</li>
    <li>php83Packages.phing</li>
    <li>php83Packages.phive</li>
    <li>php83Packages.php-codesniffer</li>
    <li>php83Packages.php-cs-fixer</li>
    <li>php83Packages.php-parallel-lint</li>
    <li>php83Packages.phpinsights</li>
    <li>php83Packages.phpmd</li>
    <li>php83Packages.phpstan</li>
    <li>php83Packages.psalm</li>
    <li>php83Packages.psysh</li>
    <li>phpactor</li>
    <li>phpdocumentor</li>
    <li>phpunit</li>
    <li>pixelfed</li>
    <li>roave-backward-compatibility-check</li>
    <li>robo</li>
    <li>rss-bridge-cli</li>
    <li>signaturepdf</li>
    <li>simplesamlphp</li>
    <li>vimPlugins.phpactor</li>
    <li>wp-cli</li>
    <li>yle-dl</li>
    <li>yle-dl.dist</li>
  </ul>
</details>

vcunat reviewed (changes requested):
The soap part is still failing on *-linux (ofBorg and locally).
```
> FAILED TEST SUMMARY
> ---------------------------------------------------------------------
> SOAP Bug #71610 - Type Confusion Vulnerability - SOAP / make_http_soap_request() [tests/bug71610.phpt]
> Bug #76348 (WSDL_CACHE_MEMORY causes Segmentation fault) [tests/bugs/bug76348.phpt]
```

vcunat committed:
phpExtensions.soap: back to doCheck = false (except on Darwin)

On *-linux we get:
> FAILED TEST SUMMARY
> ---------------------------------------------------------------------
> SOAP Bug #71610 - Type Confusion Vulnerability - SOAP / make_http_soap_request() [tests/bug71610.phpt]
> Bug #76348 (WSDL_CACHE_MEMORY causes Segmentation fault) [tests/bugs/bug76348.phpt]
Logs e.g.: https://github.com/NixOS/nixpkgs/pull/330299/checks?check_run_id=27989364591

vcunat commented:
Oh, you enabled tests on the soap part.  They pass on aarch64-darwin for me, but clearly not on *-linux, so let me simply disable them on Linux again.

reckenrode commented:
> Oh, you enabled tests on the soap part. They pass on aarch64-darwin for me, but clearly not on *-linux, so let me simply disable them on Linux again.

I assumed it was a libxml2 issue not some other issue. Thanks for catching that and fixing it.

vcunat commented:
I wouldn't say I "fixed" it, but seems good enough to merge, even though PHP 8.1 looks broken.
